src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -4683,14 +4735,22 @@ void wm_vuldet_schedule_updates(update_node **updates) {
 
     // Check updates for all the feeds
     for (int os = 0; os < OS_SUPP_SIZE; os++) {
-        if (updates[os] && wm_vuldet_check_update_period(updates[os]) && !VU_TASKS_QUEUE_FEED_UPDATE_FLAGS[updates[os]->dist_ref]) {
-            // TODO modify the flags with the proper callback
-            VU_TASKS_QUEUE_FEED_UPDATE_FLAGS[updates[os]->dist_ref] = 1;
-            vu_task_ctx_t *task = create_task(OS_INVALID, VU_INTERVAL_FEED_UPDATE, WM_TASK_PENDING, updates[os]->dist_ref, OS_INVALID);
-            if (OS_INVALID == register_task(task)) {
-                mwarn("Couldn't establish communication with task manager. There's no ID for task");
+        if (updates[os] && wm_vuldet_check_update_period(updates[os])) {
+            vu_interval_feed_update_t feed_flag = get_vendor_feed_flag(updates[os]->dist_ref);
+            if (feed_flag < VU_INTERVAL_FEED_UPDATE_SIZE &&
+                !VU_INTERVAL_FEED_UPDATE_FLAGS[get_vendor_feed_flag(updates[os]->dist_ref)]) {
```

Comment:
We could replace `get_vendor_feed_flag(updates[os]->dist_ref)` with `feed_flag`

Reply:
Done